src/workerd/io/worker.c++

```diff
@@ -160,9 +164,49 @@ void addExceptionToTrace(jsg::Lock& js,
   kj::String message;
   KJ_IF_SOME(m, error.message) {
     message = kj::str(m);
+  } else {
+    // This doesn't appear to be an Error object. Fall back to stringifying the whole value as
+    // the message.
+    if (!js.v8Isolate->IsExecutionTerminating()) {
+      v8::TryCatch tryCatch(js.v8Isolate);
+      try {
+        message = exception.toString(js);
+      } catch (jsg::JsExceptionThrown&) {
+        // Failed to stringify.
+        //
+        // Note that we're intentionally not checking tryCatch.CanContinue() here, because we still
+        // want to continue even if the isolate has been terminated.
+      }
+    }
+  }
+
+  kj::Maybe<kj::String> stack;
+  KJ_IF_SOME(s, error.stack) {
+    kj::StringPtr slice = s;
+
+    // Normally `error.stack` repeats the error type and message first. We don't want send two
```

Comment:
@kentonv — see internal thread on this, I think by stripping out here to be efficient in what gets sent across, we ended up making it such that the `stack` property in the Tail Worker ends up not including full stack trace?

Reply:
I don't know which internal thread you mean, and I don't see how the code here could possibly be stripping any part of the actual stack trace.